src/services/healthcheck.service.js

```diff
@@ -1,80 +1,134 @@
-import { config as hathorLibConfig, healthApi, txMiningApi } from '@hathor/wallet-lib';
+import { healthApi, txMiningApi, config as hathorLibConfig } from '@hathor/wallet-lib';
+import { Healthcheck, HealthcheckInternalComponent, HealthcheckHTTPComponent, HealthcheckCallbackResponse, HealthcheckStatus } from 'hathor-healthcheck-lib';
+import { initializedWallets } from './wallets.service';
+import { getConfig } from '../settings';
 
-const { buildComponentHealthCheck } = require('../helpers/healthcheck.helper');
 const { friendlyWalletState } = require('../helpers/constants');
 
-const healthService = {
+class HealthService {
+  constructor(walletIds, includeFullnode, includeTxMiningService) {
```

Comment:
note(non-blocking): This constructor is making a lot of work behind the scenes and calling other functions.

As of now, all procedures are synchronous, but should any of them be refactored to `async`, we would need to change the way we build the HealthService.

Anticipating this possibility, I leave [this SO answer](https://stackoverflow.com/a/43433773/1123007) that has helped me before on a refactoring of this kind, mentioning patterns that involve `init()` class methods and `create()` static methods.

Reply:
Thanks for the reference. I think needing to turn the methods into `async` is a remote possibility.

Even if we do need, we could just move them to an `async init()` and call it from the `healthcheck.controller.js`, which is already `async` as well.

But since you're considering this as non-blocking, I'll leave this refactor for when we actually need it.